### PR TITLE
Add batch_delete APIs and functionality

### PR DESCRIPTION
**Pull request by G-D-Petrov**
#### Reference Issues/PRs
Monday ticket ref: 7855100642

Original PR to master: https://github.com/man-group/ArcticDB/pull/2425

#### What does this implement or fix?
* Refactors `tombstone_version` into an async variant of it
* Implements batch delete functionality on top of `tombstone_versions_async`
* Adds tests for the new functionality

#### Checklist

<details>
  <summary>
   Checklist for code changes...
  </summary>
 
 - [ ] Have you updated the relevant docstrings, documentation and copyright notice?
 - [ ] Is this contribution tested against [all ArcticDB's features](../docs/mkdocs/docs/technical/contributing.md)?
 - [ ] Do all exceptions introduced raise appropriate [error messages](https://docs.arcticdb.io/error_messages/)?
 - [ ] Are API changes highlighted in the PR description?
 - [ ] Is the PR labelled as enhancement or bug so it appears in autogenerated release notes?
</details>

### CODE REVIEW

File: python/tests/unit/arcticdb/version_store/test_deletion_batch.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    assert res[0].error_category == ErrorCategory.MISSING_DATA
+    assert "version 1" in res[0].exception_string
+    assert res[1] is None
+
+    # sym1 should still be accessible
+    # sym2 should be deleted
+    assert_frame_equal(lib.read("sym1").data, df1)
+    assert len(lib.list_versions("sym1")) == 1
+    assert len(lib.list_versions("sym2")) == 0
+    assert lib.list_symbols() == ["sym1"]
```
Comment (grusev) [inline python/tests/unit/arcticdb/version_store/test_deletion_batch.py:201]: idea: add tests for for empty arrays and None for symbol (not sure if covered otherwise) ie - None, [], [sym1, None]

File: python/tests/integration/arcticdb/test_arctic_batch.py [hunk 2]
```diff
@@ -497,6 +498,150 @@ def test_delete_batch(library_factory, sym):
     assert not lib_tool.find_keys_for_id(KeyType.TABLE_DATA, sym)
 
 
+@pytest.mark.storage
+def test_delete_batch_comprehensive(arctic_library):
```
Comment (grusev) [inline python/tests/integration/arcticdb/test_arctic_batch.py:502]: repeated unfinished?
